In [21]:
import pandas as pd
hospitals = pd.read_csv('../data/Hospitals.csv')
#removing column truncation so that all columns are visualized
pd.set_option('display.max_columns', None)


In [18]:
hospitals.head()

X             Y  OBJECTID        ID  \
0 -1.331889e+07  4.346975e+06         1   5793230   
1 -1.322651e+07  4.049626e+06         2  53391362   
2 -1.315620e+07  4.031978e+06         3  11190023   
3 -1.317190e+07  4.041752e+06         4  17090028   
4 -1.313208e+07  4.037270e+06         5  23691706   

                                                 NAME  \
0                     CENTRAL VALLEY GENERAL HOSPITAL   
1  LOS ROBLES HOSPITAL & MEDICAL CENTER - EAST CAMPUS   
2                   EAST LOS ANGELES DOCTORS HOSPITAL   
3           SOUTHERN CALIFORNIA HOSPITAL AT HOLLYWOOD   
4                       KINDRED HOSPITAL BALDWIN PARK   

                     ADDRESS             CITY STATE    ZIP           ZIP4  \
0    1025 NORTH DOUTY STREET          HANFORD    CA  93230  NOT AVAILABLE   
1             150 VIA MERIDA  WESTLAKE VILAGE    CA  91362  NOT AVAILABLE   
2    4060 WHITTIER BOULEVARD      LOS ANGELES    CA  90023  NOT AVAILABLE   
3     6245 DE LONGPRE AVENUE        HOLLYWOOD    CA  90028  NOT AVAILABLE   
4  14148 FRANCISQUITO AVENUE     BALDWIN PARK    CA  91706  NOT AVAILABLE   

        TELEPHONE                TYPE  STATUS  POPULATION       COUNTY  \
0   NOT AVAILABLE  GENERAL ACUTE CARE  CLOSED          49        KINGS   
1   NOT AVAILABLE  GENERAL ACUTE CARE    OPEN          62      VENTURA   
2   NOT AVAILABLE  GENERAL ACUTE CARE    OPEN         127  LOS ANGELES   
3  (323) 462-2271  GENERAL ACUTE CARE    OPEN         100  LOS ANGELES   
4   NOT AVAILABLE  GENERAL ACUTE CARE    OPEN          95  LOS ANGELES   

  COUNTYFIPS COUNTRY   LATITUDE   LONGITUDE  NAICS_CODE  \
0      06031     USA  36.336159 -119.645667      622110   
1      06111     USA  34.154939 -118.815736      622110   
2      06037     USA  34.023647 -118.184165      622110   
3      06037     USA  34.096391 -118.325235      622110   
4      06037     USA  34.063039 -117.967438      622110   

                               NAICS_DESC  \
0  GENERAL MEDICAL AND SURGICAL HOSPITALS   
1  GENERAL MEDICAL AND SURGICAL HOSPITALS   
2  GENERAL MEDICAL AND SURGICAL HOSPITALS   
3  GENERAL MEDICAL AND SURGICAL HOSPITALS   
4  GENERAL MEDICAL AND SURGICAL HOSPITALS   

                                              SOURCE           SOURCEDATE  \
0  http://www.oshpd.ca.gov/hid/facility-listing.html  2016/02/04 00:00:00   
1  http://www.oshpd.ca.gov/hid/facility-listing.html  2019/08/02 00:00:00   
2  http://www.oshpd.ca.gov/hid/facility-listing.html  2019/08/02 00:00:00   
3  http://www.oshpd.ca.gov/hid/facility-listing.html  2019/08/02 00:00:00   
4  http://www.oshpd.ca.gov/hid/facility-listing.html  2019/08/02 00:00:00   

      VAL_METHOD             VAL_DATE                           WEBSITE  \
0  IMAGERY/OTHER  2014/02/10 00:00:00      http://www.hanfordhealth.com   
1  IMAGERY/OTHER  2014/02/10 00:00:00  http://www.losrobleshospital.com   
2  IMAGERY/OTHER  2014/02/10 00:00:00             http://www.elalax.com   
3  IMAGERY/OTHER  2014/02/10 00:00:00         http://sch-hollywood.com/   
4  IMAGERY/OTHER  2014/02/10 00:00:00      http://www.khbaldwinpark.com   

        STATE_ID                                   ALT_NAME  ST_FIPS  \
0  NOT AVAILABLE                              NOT AVAILABLE        6   
1  NOT AVAILABLE                              NOT AVAILABLE        6   
2  NOT AVAILABLE                              NOT AVAILABLE        6   
3  NOT AVAILABLE  HOLLYWOOD COMMUNITY HOSPITAL OF HOLLYWOOD        6   
4  NOT AVAILABLE                              NOT AVAILABLE        6   

         OWNER  TTL_STAFF  BEDS         TRAUMA HELIPAD  
0  PROPRIETARY       -999    49  NOT AVAILABLE       N  
1  PROPRIETARY       -999    62  NOT AVAILABLE       N  
2  PROPRIETARY       -999   127  NOT AVAILABLE       N  
3  PROPRIETARY       -999   100  NOT AVAILABLE       N  
4  PROPRIETARY       -999    95  NOT AVAILABLE       N

In [19]:
hospitals.tail()

X             Y  OBJECTID         ID  \
7591 -8.566905e+06  4.728815e+06      7607  183520904   
7592 -1.219624e+07  3.676931e+06      7608  192185607   
7593 -8.084677e+06  5.094112e+06      7609  196706457   
7594 -8.522423e+06  4.718512e+06      7610  191021401   
7595 -1.244921e+07  4.976468e+06      7611  196784108   

                                               NAME  \
7591  ADVENTIST HEALTHCARE WHITE OAK MEDICAL CENTER   
7592      COPPER QUEEN DOUGLAS EMERGENCY DEPARTMENT   
7593                      WHITING FORENSIC HOSPITAL   
7594             J KENT MCNEW FAMILY MEDICAL CENTER   
7595                  UNIVERSITY ORTHOPAEDIC CENTER   

                         ADDRESS            CITY STATE    ZIP           ZIP4  \
7591           11890 HEALING WAY   SILVER SPRING    MD  20904  NOT AVAILABLE   
7592           100 E. 5TH STREET         DOUGLAS    AZ  85607  NOT AVAILABLE   
7593                70 OBRIEN DR      MIDDLETOWN    CT   6457           3945   
7594  175 HARRY S TRUMAN PARKWAY       ANNAPOLIS    MD  21401  NOT AVAILABLE   
7595              590 WAKARA WAY  SALT LAKE CITY    UT  84108  NOT AVAILABLE   

           TELEPHONE                TYPE STATUS  POPULATION        COUNTY  \
7591  (301) 891-5651  GENERAL ACUTE CARE   OPEN         178    MONTGOMERY   
7592  (520) 364-7659  GENERAL ACUTE CARE   OPEN        -999       COCHISE   
7593  (860) 262-5400         PSYCHIATRIC   OPEN         229     MIDDLESEX   
7594  (667) 204-7300         PSYCHIATRIC   OPEN        -999  ANNE ARUNDEL   
7595  (801) 587-7109  GENERAL ACUTE CARE   OPEN        -999     SALT LAKE   

     COUNTYFIPS COUNTRY   LATITUDE   LONGITUDE  NAICS_CODE  \
7591      24031     USA  39.049854  -76.957821      622110   
7592      04003     USA  31.339594 -109.560682      622110   
7593      09007     USA  41.552040  -72.625890      622210   
7594      24003     USA  38.977938  -76.558229      622210   
7595      49035     USA  40.756339 -111.833184      622110   

                                     NAICS_DESC  \
7591     GENERAL MEDICAL AND SURGICAL HOSPITALS   
7592     GENERAL MEDICAL AND SURGICAL HOSPITALS   
7593  PSYCHIATRIC AND SUBSTANCE ABUSE HOSPITALS   
7594  PSYCHIATRIC AND SUBSTANCE ABUSE HOSPITALS   
7595     GENERAL MEDICAL AND SURGICAL HOSPITALS   

                                                           SOURCE  \
7591  http://dhmh.maryland.gov/ohcq/pages/licensee-directory.aspx   
7592                          http://www.azdhs.gov/als/databases/   
7593       https://www.elicense.ct.gov/lookup/generateroster.aspx   
7594  http://dhmh.maryland.gov/ohcq/pages/licensee-directory.aspx   
7595      http://gis.utah.gov/data/health/health-care-facilities/   

               SOURCEDATE     VAL_METHOD             VAL_DATE  \
7591  2019/09/03 00:00:00  IMAGERY/OTHER  2020/05/05 00:00:00   
7592  2019/09/17 00:00:00  IMAGERY/OTHER  2020/05/05 00:00:00   
7593  2019/08/15 00:00:00  IMAGERY/OTHER  2020/05/08 00:00:00   
7594  2019/09/03 00:00:00  IMAGERY/OTHER  2020/05/05 00:00:00   
7595  2019/10/21 00:00:00  IMAGERY/OTHER  2020/05/05 00:00:00   

                                                                                                                                                    WEBSITE  \
7591  https://www.adventisthealthcare.com/locations/profile/white-oak-medical-center/?utm_source=local-listing&utm_medium=organic&utm_campaign=website-link   
7592                                                                                                https://cqch.org/cqch-clinics/copper-queen-douglas-rhc/   
7593                                                                                              https://portal.ct.gov/dmhas/wfh/whiting-forensic-hospital   
7594                                                                                        https://aahs.org/locations/j--kent-mcnew-family-medical-center/   
7595                                                                                                              http

Not all of the hospitals are in Tennessee, they are from all over the US. Something is wrong with the TTL_Staff column since it gives the same value of -999 for all visible rows. Also its not clear what columns 'X' and 'Y' are referring to, perhaps latitute and longitude.

In [14]:
hospitals.shape

(7596, 34)

There are 7596 rows and 34 columns in the hospitals dataframe.

In [15]:
hospitals.columns

Index(['X', 'Y', 'OBJECTID', 'ID', 'NAME', 'ADDRESS', 'CITY', 'STATE', 'ZIP',
       'ZIP4', 'TELEPHONE', 'TYPE', 'STATUS', 'POPULATION', 'COUNTY',
       'COUNTYFIPS', 'COUNTRY', 'LATITUDE', 'LONGITUDE', 'NAICS_CODE',
       'NAICS_DESC', 'SOURCE', 'SOURCEDATE', 'VAL_METHOD', 'VAL_DATE',
       'WEBSITE', 'STATE_ID', 'ALT_NAME', 'ST_FIPS', 'OWNER', 'TTL_STAFF',
       'BEDS', 'TRAUMA', 'HELIPAD'],
      dtype='object')

In [49]:
sliced_hospitals = hospitals[['NAME', 'ADDRESS', 'CITY', 'STATE', 'TYPE', 'POPULATION', 'COUNTY', 'COUNTYFIPS', 'NAICS_DESC', 'BEDS']]
sliced_hospitals

NAME  \
0     CENTRAL VALLEY GENERAL HOSPITAL                      
1     LOS ROBLES HOSPITAL & MEDICAL CENTER - EAST CAMPUS   
2     EAST LOS ANGELES DOCTORS HOSPITAL                    
3     SOUTHERN CALIFORNIA HOSPITAL AT HOLLYWOOD            
4     KINDRED HOSPITAL BALDWIN PARK                        
...                             ...                        
7591  ADVENTIST HEALTHCARE WHITE OAK MEDICAL CENTER        
7592  COPPER QUEEN DOUGLAS EMERGENCY DEPARTMENT            
7593  WHITING FORENSIC HOSPITAL                            
7594  J KENT MCNEW FAMILY MEDICAL CENTER                   
7595  UNIVERSITY ORTHOPAEDIC CENTER                        

                         ADDRESS             CITY STATE                TYPE  \
0     1025 NORTH DOUTY STREET     HANFORD          CA    GENERAL ACUTE CARE   
1     150 VIA MERIDA              WESTLAKE VILAGE  CA    GENERAL ACUTE CARE   
2     4060 WHITTIER BOULEVARD     LOS ANGELES      CA    GENERAL ACUTE CARE   
3     6245 DE LONGPRE AVENUE      HOLLYWOOD        CA    GENERAL ACUTE CARE   
4     14148 FRANCISQUITO AVENUE   BALDWIN PARK     CA    GENERAL ACUTE CARE   
...                         ...            ...     ..                   ...   
7591  11890 HEALING WAY           SILVER SPRING    MD    GENERAL ACUTE CARE   
7592  100 E. 5TH STREET           DOUGLAS          AZ    GENERAL ACUTE CARE   
7593  70 OBRIEN DR                MIDDLETOWN       CT    PSYCHIATRIC          
7594  175 HARRY S TRUMAN PARKWAY  ANNAPOLIS        MD    PSYCHIATRIC          
7595  590 WAKARA WAY              SALT LAKE CITY   UT    GENERAL ACUTE CARE   

      POPULATION        COUNTY COUNTYFIPS  \
0     49          KINGS         06031       
1     62          VENTURA       06111       
2     127         LOS ANGELES   06037       
3     100         LOS ANGELES   06037       
4     95          LOS ANGELES   06037       
...   ..                  ...     ...       
7591  178         MONTGOMERY    24031       
7592 -999         COCHISE       04003       
7593  229         MIDDLESEX     09007       
7594 -999         ANNE ARUNDEL  24003       
7595 -999         SALT LAKE     49035       

                                     NAICS_DESC  BEDS  
0     GENERAL MEDICAL AND SURGICAL HOSPITALS     49    
1     GENERAL MEDICAL AND SURGICAL HOSPITALS     62    
2     GENERAL MEDICAL AND SURGICAL HOSPITALS     127   
3     GENERAL MEDICAL AND SURGICAL HOSPITALS     100   
4     GENERAL MEDICAL AND SURGICAL HOSPITALS     95    
...                                      ...     ..    
7591  GENERAL MEDICAL AND SURGICAL HOSPITALS     178   
7592  GENERAL MEDICAL AND SURGICAL HOSPITALS    -999   
7593  PSYCHIATRIC AND SUBSTANCE ABUSE HOSPITALS  229   
7594  PSYCHIATRIC AND SUBSTANCE ABUSE HOSPITALS -999   
7595  GENERAL MEDICAL AND SURGICAL HOSPITALS    -999   

[7596 rows x 10 columns]

In [40]:
#The following line modifies the dataframe so that the following columns are permanently removed from the dataset, which was not asked for so I commented it out.
#hospitals.drop(columns = ['X', 'Y', 'OBJECTID', 'ID', 'ZIP', 'ZIP4', 'TELEPHONE', 'STATUS', 'COUNTRY', 'LATITUDE', 'LONGITUDE', 'NAICS_CODE', 'SOURCE', 'SOURCEDATE', 'VAL_METHOD', 'VAL_DATE', 'WEBSITE', 'STATE_ID', 'ALT_NAME', 'ST_FIPS', 'ST_FIPS', 'OWNER', 'TTL_STAFF', 'TRAUMA', 'HELIPAD' ])


In [50]:
sliced_hospitals['TYPE'].value_counts()

GENERAL ACUTE CARE    4481
CRITICAL ACCESS       1027
PSYCHIATRIC           759 
LONG TERM CARE        431 
REHABILITATION        378 
MILITARY              216 
SPECIAL               145 
CHILDREN              130 
WOMEN                 20  
CHRONIC DISEASE       9   
Name: TYPE, dtype: int64

The most common type of hospital in the dataframe is General Acute Care.

In [56]:
nashville_hospitals = sliced_hospitals.loc[((sliced_hospitals["CITY"] == "NASHVILLE") & (sliced_hospitals['STATE'] == 'TN'))]
nashville_hospitals

NAME  \
545   TRISTAR SKYLINE MEDICAL CENTER                  
3449  SELECT SPECIALTY HOSPITAL - NASHVILLE           
3485  CURAHEALTH NASHVILLE                            
5429  THE WOMEN'S HOSPITAL AT CENTENNIAL              
5433  METROPOLITAN NASHVILLE GENERAL HOSPITAL         
6289  SAINT THOMAS MIDTOWN HOSPITAL                   
6293  VANDERBILT STALLWORTH REHABILITATION HOSPITAL   
6661  VA MEDICAL CENTER - NASHVILLE CAMPUS            
6774  SAINT THOMAS HOSPITAL FOR SPECIALTY SURGERY     
6777  SAINT THOMAS WEST HOSPITAL                      
6786  TRISTAR SOUTHERN HILLS MEDICAL CENTER           
6808  TRISTAR CENTENNIAL MEDICAL CENTER               
6813  VANDERBILT UNIVERSITY HOSPITAL                  

                             ADDRESS       CITY STATE                TYPE  \
545   3441 DICKERSON PIKE             NASHVILLE  TN    GENERAL ACUTE CARE   
3449  2000 HAYES STREET, SUITE 500    NASHVILLE  TN    LONG TERM CARE       
3485  1412 COUNTY HOSPITAL ROAD, B-1  NASHVILLE  TN    LONG TERM CARE       
5429  2221 MURPHY AVENUE              NASHVILLE  TN    WOMEN                
5433  1818 ALBION STREET              NASHVILLE  TN    GENERAL ACUTE CARE   
6289  2000 CHURCH STREET              NASHVILLE  TN    GENERAL ACUTE CARE   
6293  2201 CHILDREN'S WAY             NASHVILLE  TN    REHABILITATION       
6661  1310 24TH AVE S                 NASHVILLE  TN    MILITARY             
6774  2011 MURPHY AVENUE              NASHVILLE  TN    GENERAL ACUTE CARE   
6777  4220 HARDING ROAD               NASHVILLE  TN    GENERAL ACUTE CARE   
6786  391 WALLACE ROAD                NASHVILLE  TN    GENERAL ACUTE CARE   
6808  2300 PATTERSON STREET           NASHVILLE  TN    GENERAL ACUTE CARE   
6813  1211 MEDICAL CENTER DRIVE       NASHVILLE  TN    GENERAL ACUTE CARE   

      POPULATION    COUNTY COUNTYFIPS  \
545   213         DAVIDSON  47037       
3449  70          DAVIDSON  47037       
3485  60          DAVIDSON  47037       
5429 -999         DAVIDSON  47037       
5433  150         DAVIDSON  47037       
6289  683         DAVIDSON  47037       
6293  80          DAVIDSON  47037       
6661  493         DAVIDSON  47037       
6774  23          DAVIDSON  47037       
6777  541         DAVIDSON  47037       
6786  126         DAVIDSON  47037       
6808  733         DAVIDSON  47037       
6813  1091        DAVIDSON  47037       

                                                        NAICS_DESC  BEDS  
545   GENERAL MEDICAL AND SURGICAL HOSPITALS                        213   
3449  SPECIALTY (EXCEPT PSYCHIATRIC AND SUBSTANCE ABUSE) HOSPITALS  70    
3485  SPECIALTY (EXCEPT PSYCHIATRIC AND SUBSTANCE ABUSE) HOSPITALS  60    
5429  SPECIALTY (EXCEPT PSYCHIATRIC AND SUBSTANCE ABUSE) HOSPITALS -999   
5433  GENERAL MEDICAL AND SURGICAL HOSPITALS                        150   
6289  GENERAL MEDICAL AND SURGICAL HOSPITALS                        683   
6293  REHABILITATION HOSPITALS (EXCEPT ALCOHOLISM, DRUG ADDICTION)  80    
6661  GENERAL MEDICAL AND SURGICAL HOSPITALS                        493   
6774  GENERAL MEDICAL AND SURGICAL HOSPITALS                        23    
6777  GENERAL MEDICAL AND SURGICAL HOSPITALS                        541   
6786  GENERAL MEDICAL AND SURGICAL HOSPITALS                        126   
6808  GENERAL MEDICAL AND SURGICAL HOSPITALS                        733   
6813  GENERAL MEDICAL AND SURGICAL HOSPITALS                        1091

Why is there one hospital with -999 beds? got to be an error with formatting in the original csv file.

In [60]:
tennessee_hospitals = sliced_hospitals.loc[sliced_hospitals['STATE'] == 'TN']
tennessee_hospitals['COUNTY'].value_counts()

SHELBY        23
DAVIDSON      15
HAMILTON      13
KNOX          11
WASHINGTON    6 
             .. 
PUTNAM        1 
CUMBERLAND    1 
JOHNSON       1 
MARSHALL      1 
ROBERTSON     1 
Name: COUNTY, Length: 80, dtype: int64

Since this shows that there are COUNTY values for 80 rows, that means that 15 counties in TN don't have any hospitals.